R/sensors.R

```diff
@@ -27,6 +27,79 @@ sensor_features <- function(sensor_data, transform, extract, extract_on, groups)
+#' tapping measurements. buttonid can be from c('TappedButtonLeft','TappedButtonRight','TappedButtonNone') 
+#' indicating a tap that has been classified as to the left, right or neither of those places on the screen
+#' @param depressThr A numerical threshold for intertap distance in x axis
+#' @return A dataframe of features.
+tapping_features <- function(tap_data,
+                             depressThr = 20) {
+  
+  results <- GetLeftRightEventsAndTapIntervals(tapData = tap_data,
+                                               depressThr = depressThr)
+  tapInter <- results$tapInter
+  tapData <- results$tapData
+  error <- results$error
+  
+  # check error - if after cleaning tapping data less than 5 data points remain
+  if (error == TRUE) {
+    tapFeatures <- dplyr::tibble(error = "post cleaning less than 5 tap points remain")
+    return(tapFeatures)
+  }
+  
+  meanX <- mean(tapData$x)
+  iL <- tapData$x < meanX
+  iR <- tapData$x >= meanX
+  driftLeft <- calculateDrift(x = tapData[iL,"x"], y = tapData[iL, "y"])
+  driftRight <- calculateDrift(x = tapData[iR,"x"], y = tapData[iR, "y"])
+
+  intertap_features <- intertap_summary_features(tapInter = tapInter)
+  if(intertap_features$error == 'None'){
+    intertap_features <- intertap_features %>% dplyr::select(-error)
+    colnames(intertap_features) <- paste0(colnames(intertap_features),'TapInter')
+  }else{
+    colnames(intertap_features) <- paste0(colnames(intertap_features),'TapInter')
+  }
+
+  tapdrift_left_features <- tapdrift_summary_features(tapDrift = driftLeft)
+  if(tapdrift_left_features$error == 'None'){
+    tapdrift_left_features <- tapdrift_left_features %>% dplyr::select(-error)
+    colnames(tapdrift_left_features) <- paste0(colnames(tapdrift_left_features),'DriftLeft')
+  }else{
+    colnames(tapdrift_left_features) <- paste0(colnames(tapdrift_left_features),'DriftLeft')
+  }
+  
+  tapdrift_right_features <- tapdrift_summary_features(tapDrift = driftRight)
+  if(tapdrift_right_features$error == 'None'){
+    tapdrift_right_features <- tapdrift_right_features %>% dplyr::select(-error)
+    colnames(tapdrift_right_features) <- paste0(colnames(tapdrift_right_features),'DriftRight')
+  }else{
+    colnames(tapdrift_right_features) <- paste0(colnames(tapdrift_right_features),'DriftRight')
+  }
+  
+  tapdata_features <- tap_data_summary_features(tapData = tap_data)
+  if(tapdata_features$error == 'None'){
+    tapdata_features <- tapdata_features %>% dplyr::select(-error)
+  }
+  
+  tapFeatures <- dplyr::bind_cols(intertap_features,
+                           tapdrift_left_features,
+                           tapdrift_right_features,
+                           tapdata_features)
+  
+  ftrs_error <- grep('error', colnames(tapFeatures))
+  ftrs_error <- paste(tapFeatures[ftrs_error], collapse = ' ; ')
+  if(ftrs_error == ''){
+    ftrs_error = 'None'
+  }
+  
+  tapFeatures$error <- ftrs_error
+  return(tapFeatures %>% as.data.frame())
```

Comment:
Move the piping function up one line and just return a single object with no "function stuff" within the return statement.